bazel/generate_objc.bzl

```diff
@@ -0,0 +1,243 @@
+load(
+    "//bazel:protobuf.bzl",
+    "get_include_protoc_args",
+    "get_plugin_args",
+    "get_proto_root",
+    "proto_path_to_generated_filename",
+)
+load(
+    ":grpc_util.bzl", 
+    "to_upper_camel_with_extension",
+    "label_to_file",
+)
+
+_GRPC_PROTO_HEADER_FMT = "{}.pbrpc.h"
+_GRPC_PROTO_SRC_FMT = "{}.pbrpc.m"
+_PROTO_HEADER_FMT = "{}.pbobjc.h"
+_PROTO_SRC_FMT = "{}.pbobjc.m"
+_GENERATED_PROTOS_DIR = "_generated_protos"
+
+def _generate_objc_impl(ctx):
+
+    """Implementation of the generate_cc rule."""
+    protos = [
+        f
+        for src in ctx.attr.srcs
+        for f in src[ProtoInfo].transitive_imports.to_list()
+    ]
+    outs = []
+    proto_root = get_proto_root(
+        ctx.label.workspace_root,
+    )
+
+    label_package = _join_directories([ctx.label.workspace_root, ctx.label.package])
+
+    files_with_rpc = [label_to_file(f) for f in ctx.attr.files_with_service]
+    for proto in protos:
+        outs += [
+            proto_path_to_generated_filename(
+                _GENERATED_PROTOS_DIR + "/" +
+                _get_directory_from_proto(proto) + _get_slash_from_proto(proto) +
```

Comment:
@muxi `_get_slash_frrom_proto` potentially returns an empty string (file is in root dir). I know it's a dumb name but because `_get_directory_from_proto` is used elsewhere without the slash so we can't simply replace or combine them. Maybe suggest a better name for it?

Reply:
I see. I can't come up with a better name too :joy:. Maybe `_get_slash_or_null_from_proto`, and comment in the function that why we can't just return `"/"`.